spotify-api/spotifyRequests.ts

```diff
@@ -206,3 +206,216 @@ export const fetchArtistImagesBySpotifyId = async (
+          const retryDelay = parseInt(err.response!.headers['retry-after'] as string) * 1000;
+          //   console.log(`Request failed. Retrying in ${retryDelay / 1000} seconds...`);
+          return new Promise((resolve) => {
+            setTimeout(() => {
+              resolve(
+                fetchSingleSpotifyTrackByYoutubePlaylistURL(accessToken, playlistTitle, maxRetries, currentRetry + 1)
+              );
+            }, retryDelay);
+          });
+        } else {
+          // Max retries reached,
+          // console.log(`Max tries exceeded. Can't get song with title ${playlistTitle.title}`);
+          return {
+            youtube: {
+              name: playlistTitle.title,
+            },
+          };
+        }
+      } else {
+        // Can't find a correspond track for the youtube video
+        // console.log(`Can't get song with title ${playlistTitle.title}`);
+        return {
+          youtube: {
+            name: playlistTitle.title,
+          },
+        };
+      }
+    });
+};
+
+function chunkArray<T>(arr: T[], chunkSize: number) {
+  const result = [];
+  for (let i = 0; i < arr.length; i += chunkSize) {
+    result.push(arr.slice(i, i + chunkSize));
+  }
+  return result;
+}
+
+export const fetchSpotifyTracksByYoutubePlaylistID = (
+  accessToken: string,
+  youtubePlaylistID: string
+): Promise<spotifyYoutubeTrackMapping[]> => {
+  return axios
+    .get<{ id: string; playlistTitles: { title: string }[] }>(
+      queryParamsStringify(`${API_KEY}/playlist`, {
+        id: youtubePlaylistID,
+      })
+    )
+    .then(async (response) => {
+      // batch requests, don't do all at once
+      const BATCH_SIZE = 50;
+      const partitions = chunkArray(response.data.playlistTitles, BATCH_SIZE);
+
+      let spotifyTrackInfoForTitles: spotifyYoutubeTrackMapping[] = [];
+      for (const chunk of partitions) {
+        //  fullfill the 50 promises before going onto the next batch
+        const fetchIntermediateResult = await Promise.all(
+          // todo: Do some preprocessing on the titleOnPlaylist string
+          // for example, remove words that grep with (official music video)
+          chunk.map((titleOnPlaylist) => fetchSingleSpotifyTrackByYoutubePlaylistURL(accessToken, titleOnPlaylist))
+        );
+        spotifyTrackInfoForTitles.push(...fetchIntermediateResult);
+      }
+
+      return spotifyTrackInfoForTitles;
+    })
+    .catch((error) => {
+      // console.log('Possible error from external service.');
+      throw error;
+    });
+};
+
+export interface createSpotifyPlaylistResponse {
+  spotifyPlaylistID: string;
+  spotifyPlaylistURL: string;
+}
+const createSpotifyPlaylist = (accessToken: string, userSpotifyID: string): Promise<createSpotifyPlaylistResponse> => {
+  return axios
+    .post<SpotifyApi.CreatePlaylistResponse>(
+      `https://api.spotify.com/v1/users/${userSpotifyID}/playlists`,
+      {
+        // todo: get the actual youtube playlist name
```

Comment:
I think this would be nice to have, if you can get it in that would be great

Reply:
Another PR, not entirely sure how to get this from youtube yet.